extensions/amp-sidebar/0.1/amp-sidebar.js

```diff
@@ -381,21 +383,21 @@ export class AmpSidebar extends AMP.BaseElement {
       this.scheduleResume(children);
       tryFocus(this.element);
       this.triggerEvent_(SidebarEvents.OPEN);
+      this.element.setAttribute('aria-hidden', 'false');
     } else {
       // On close sidebar
       this.vsync_.mutate(() => {
         toggle(this.element, /* display */false);
         this.schedulePause(this.getRealChildren());
-        // TODO(cathyxz, 12479): Re-enable with updated heuristic on whether
-        // returning focus to opener is the right choice or not. Current
-        // issue is that if an item in sidebar has a .focus, .scrollTo
-        // or just local # navigation before closing the sidebar, moving the
-        // focus would override their effects.
-        // Return focus to source element if applicable
-        // if (this.openerElement_) {
-        //   tryFocus(this.openerElement_);
-        // }
+        const sidebarIsActive =
+          this.element.querySelectorAll(':focus').length > 0;
+        const scrollDidNotChange =
+          (window./*REVIEW*/scrollY == this.initialScrollY_);
+        if (this.openerElement_ && sidebarIsActive && scrollDidNotChange) {
+          tryFocus(this.openerElement_);
+        }
         this.triggerEvent_(SidebarEvents.CLOSE);
+        this.element.setAttribute('aria-hidden', 'true');
```

Comment:
I moved this line around because we need one visual indicator to poll on to signal that the sidebar has indeed been opened or closed and the animation is complete. The animation is a debounced function with a 550ms timeout, which really doesn't play nice with polling unless there's a visible indicator to signal its end. 

I'm not sure that this is the best one, but I don't know if there is a better one. Feedback is welcome here.

Reply:
Are you refactoring this to use the `aria-hidden` attribute as a flag in test code? If that's the case, can you listen to `SidebarEvents.CLOSE` in the test instead? I think setting `aria-hidden` as soon as the animation starts might be better, since for a screen reader user they would no longer need to read that information as soon as the animation starts.

Reply:
This is a good idea. Actually we might as well toggle focus in `close` instead of `onAnimationEnd`, since for screen readers the animation isn't relevant anyway...